src/main/java/com/scalyr/logback/ScalyrAppender.java

```diff
@@ -39,75 +41,67 @@
-    }
+    public void setApiKey(String apiKey) { this.apiKey = apiKey; }
 
-    public String getServerHost() {
-        return this.serverHost == null ? "" : this.serverHost.trim();
-    }
+    public String getServerHost() { return this.serverHost == null ? "" : this.serverHost.trim(); }
 
-    public void setServerHost(String serverHost) {
-        this.serverHost = serverHost;
-    }
+    public void setServerHost(String serverHost) { this.serverHost = serverHost; }
 
-    public Integer getMaxBufferRam() {
-        return maxBufferRam;
-    }
+    public Integer getMaxBufferRam() { return maxBufferRam; }
 
-    public void setMaxBufferRam(String maxBufferRam) {
-        if(maxBufferRam != null && !"".equals(maxBufferRam)) {
-            maxBufferRam = maxBufferRam.toLowerCase().trim();
-            if(maxBufferRam.contains("m")) {
-                this.maxBufferRam = Integer.valueOf(maxBufferRam.substring(0, maxBufferRam.indexOf("m"))) * 1048576;
-            } else if (maxBufferRam.contains("k")) {
-                this.maxBufferRam = Integer.valueOf(maxBufferRam.substring(0, maxBufferRam.indexOf("k"))) * 1024;
-            } else {
-                this.maxBufferRam = Integer.valueOf(maxBufferRam);
-            }
-        }
-    }
+    public void setMaxBufferRam(String maxBufferRam) { this.maxBufferRam = Util.stringToIntMemory(maxBufferRam); }
 
-    public void setLogfile(String logfile) {
-        this.logfile = logfile;
-    }
+    public void setLogfile(String logfile) { this.logfile = logfile; }
 
-    public String getLogfile() {
-        return logfile;
-    }
+    public String getLogfile() { return logfile; }
 
-    public String getParser() {
-        return parser;
-    }
+    public String getParser() { return parser; }
 
-    public void setEnv(String env) {
-        this.env = env;
-    }
+    public void setEnv(String env) { this.env = env; }
 
-    public String getEnv() {
-        return env;
-    }
+    public String getEnv() { return env; }
 
-    public void setParser(String parser) {
-        this.parser = parser;
-    }
+  /**
+   * Use this to describe any additional server attributes. Takes a string that is kv pairs separated by a comma e.g.:
+   * <pre>
+   *   appName=appofdoom,key1=value1
+   * </pre>
+   * An example of what to put in your logback.xml:
+   <pre><code>
+   &lt;configuration&gt;
+     &lt;appender name="scalyr" class="com.scalyr.logback.ScalyrAppender"&gt;
+       &lt;apiKey&gt;YOUR KEY&lt;/apiKey&gt;
+       &lt;extraAttributes&gt;appName=appofdoom,zodiac=rooster&lt;/extraAttributes&gt;
+       ...
+     &lt;/appender&gt;
+   &lt;/configuration&gt;
+   </code></pre>
+   * <p>
+   * The extraAttributes will be appended to the serverAttributes list and will be searchable in scalyr by a query like:
+   * <pre>
+   *   $zodiac == "rooster" "hello world"
+   * </pre>
+   *
+   * @param extraAttributes String that is a JSON representation of additional key/value pairs to add
```

Comment:
Re. JSON: same comment as in ScalyrAppender.

Reply:
👍